### ISSUE TITLE
proxy crash 

### ISSUE THREAD

**Issue by jkasarherou** [open-io/oio-sds#202]
After a simple request on the proxy:

```
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f166e2ce700 (LWP 17634)]
0x00007f166e1cc9b0 in ?? ()
(gdb) bt
#0  0x00007f166e1cc9b0 in ?? ()
#1  0x00007f1670c51e2d in ?? () from /usr/lib/libglib-2.0.so.0
#2  0x000000000041ace8 in header_provider (k0=0x7f1668001c20 "Connection", v=0x7f1668001c2c "keep-alive")
    at /home/noone/public_git/github.com/jkasarherou/oio-sds/proxy/transport_http.c:624
#3  0x0000000000419216 in _manage_and_renew_header (parser=0x7f1668001a00, buf=0x102a140)
    at /home/noone/public_git/github.com/jkasarherou/oio-sds/proxy/transport_http.c:151
#4  0x00000000004193e1 in http_parse (
    data=0x7f1660001a00 "GET / HTTP/1.1\r\nHost: 127.0.0.1:6002\r\nConnection: keep-alive\r\nAccept-Encoding: gzip, deflate\r\nAccept: */*\r\nUser-Agent: HTTPie/0.9.2\r\n\r\n", available=135) at /home/noone/public_git/github.com/jkasarherou/oio-sds/proxy/transport_http.c:198
#5  0x000000000041aa20 in http_notify_input (clt=0x7f1668001a50) at /home/noone/public_git/github.com/jkasarherou/oio-sds/proxy/transport_http.c:668
#6  0x00007f1671d2f7c6 in _notify () at /home/noone/public_git/github.com/jkasarherou/oio-sds/server/network_server.c:1439
#7  0x00007f1671d2f8a0 in _client_manage_input () at /home/noone/public_git/github.com/jkasarherou/oio-sds/server/network_server.c:1463
#8  0x00007f1671d2fa61 in _client_manage_event (clt=0x7f1668001a50, events=1)
    at /home/noone/public_git/github.com/jkasarherou/oio-sds/server/network_server.c:1533
#9  0x00007f1671d2e96c in _work_on_client (srv=0x1028cc0, clt=0x7f1668001a50)
    at /home/noone/public_git/github.com/jkasarherou/oio-sds/server/network_server.c:1039
#10 0x00007f1671d2eb7d in _thread_cb_worker (td=0x1028cc0) at /home/noone/public_git/github.com/jkasarherou/oio-sds/server/network_server.c:1088
#11 0x00007f1670c4c625 in ?? () from /usr/lib/libglib-2.0.so.0
#12 0x00007f166eb2a4a4 in start_thread () from /usr/lib/libpthread.so.0
#13 0x00007f166f09913d in clone () from /usr/lib/libc.so.6
```

**Comment by jfsmig** [comment]
It seems to be a bug appearing in nested functions, compiled with gcc 5, with glib2 2.44. #206 tries to fix this.

**Comment by jkasarherou** [comment]
fixed in #206
